pyproject.toml

```diff
@@ -88,33 +88,10 @@ dependencies = [
 dynamic = ["version"]
 
 [project.optional-dependencies]
-docs = [
-    "sphinxcontrib-websupport",
-]
-lint = [
-    "ruff==0.11.7",
-    "mypy==1.15.0",
-    "sphinx-lint>=0.9",
-    "types-colorama==0.4.15.20240311",
-    "types-defusedxml==0.7.0.20240218",
-    "types-docutils==0.21.0.20241128",
-    "types-Pillow==10.2.0.20240822",
-    "types-Pygments==2.19.0.20250305",
-    "types-requests==2.32.0.20250328",  # align with requests
-    "types-urllib3==1.26.25.14",
-    "pyright==1.1.400",
-    "pytest>=8.0",
-    "pypi-attestations==0.0.25",
-    "betterproto==2.0.0b6",
-]
-test = [
-    "pytest>=8.0",
-    "pytest-xdist[psutil]>=3.4",
-    "defusedxml>=0.7.1", # for secure XML/HTML parsing
-    "cython>=3.0",
-    "setuptools>=70.0",  # for Cython compilation
-    "typing_extensions>=4.9",  # for typing_extensions.Unpack
-]
+# retained to not break installing 'sphinx[…]'
+docs = []
+lint = []
+test = []
```

Comment:
why not break these workflows? they will already not work the way users will expect, and will only affect contributors (not downstream users) anyway

Reply:
The impact here is ~1,500 repos: https://github.com/search?q=%2Fsphinx%5C%5B%28docs%7Clint%7Ctest%29%2F&type=code

The best option here is to do nothing and not break downstream. The second best option, in my opinion, is to do this PR: don't make the projects fail to resolve, but remove the redundant dependencies.

A

Reply:
It turns out `pip` and `uv` only emit a warning, so I am less concerned about fully removing the dependencies.

A